### DISCUSSION TITLE
How to properly notify about transient state?

### DISCUSSION THREAD

**Discussion by petrbrzek** [statelyai/xstate#3182] [Q&A] [answered]
For the `subscribed` state I first had `always` with the `target` to `notSubscribed`, but then I found out that I don't know about this state in the React app at all (as it's immediately redirected to `notSubscribed`), so now I'm "hacking" it with `after`. The React app reacts to the `subscribed` state by resetting a `<form>`. Can I do this in a more idiomatic way? Thanks.

```
const subscribeMachine =
  createMachine({
    id: 'Subscriber',
    initial: 'notSubscribed',
    states: {
      notSubscribed: {
        on: {
          SUBSCRIBING: {
            target: 'loading',
          },
        },
      },
      loading: {
        invoke: {
          id: 'subscribeUser',
          src: (context, event) => subscribeUser(event.email),
          onDone: {
            target: 'subscribed',
          },
          onError: {
            target: 'error',
          },
        },
      },
      error: {
        on: {
          SUBSCRIBING: { target: 'loading' },
        },
        after: {
          3000: {
            target: 'notSubscribed',
          },
        },
      },
      subscribed: {
        after: {
          1: {
            target: 'notSubscribed',
          },
        },
      },
    },
  })
```

**Comment by davidkpiano** [comment, marked as answer]
Yes; if you have a "transient state" that immediately transitions to another state, the state machine will spend zero-time in that state, so it shouldn't be reflected in the UI.

If you are performing an action in response to being in a transient state right before transitioning to a different state, that would be better represented as an action:

```js
subscribed: {
  always: {
    actions: 'doSomething',
    target: 'notSubscribed'
  }
}
```

And you can specify that action in React:

```jsx
const [state, send] = useMachine(subscribeMachine, {
  actions: {
    doSomething: () => { ... }
  }
});
```
